src/serializer/ResidualReactElementVisitor.js

```diff
@@ -111,9 +111,29 @@ export class ResidualReactElementVisitor {
       },
     });
 
-    if (this.realm.react.output === "create-element" || isReactFragment) {
-      this.someReactElement = reactElement;
+    // Our serializer requires that every value we serialize must first be visited in every scope where it appears. In
+    // our React element serializer we serialize some values (namely `React.createElement` and `React.Fragment`) that do
+    // not necessarily appear in our source code. We must manually visit these values in our visitor pass for the values
+    // to be serializable.
+    {
+      if (this.realm.react.output === "create-element") {
+        const reactLibraryObject = this._getReactLibraryValue();
+        invariant(reactLibraryObject instanceof ObjectValue);
+        const createElement = reactLibraryObject.properties.get("createElement");
+        invariant(createElement !== undefined);
+        const reactCreateElement = Get(this.realm, reactLibraryObject, "createElement");
+        // Our `createElement` value will be used in the prelude of the optimized function we serialize to initialize
+        // our hoisted React elements. So we need to ensure that we visit our value in a scope above our own to allow
+        // the function to be used in our optimized function prelude. We use our global scope to accomplish this. We are
+        // a "friend" class of `ResidualHeapVisitor` so we call one of its private methods.
+        this.residualHeapVisitor._visitInUnrelatedScope(this.residualHeapVisitor.globalGenerator, reactCreateElement);
```

Comment:
Visiting this value in the global generator is a bit clowny (read comment), but it is the same thing that the previous `fixpoint_rerun` code did.

Is this be a valid call to `_withScope`? Why? There’s a scary comment on `_withScope` I don’t yet understand which makes me less confident about this.

https://github.com/facebook/prepack/blob/e3f4262c9dafce766835acdaff39af04de231fae/src/serializer/ResidualHeapVisitor.js#L243-L245